### PR TITLE
Check buffed Muscle before using Powerful Glove and prefer Moleman for Thesis

### PR DESCRIPTION
**Pull request by eegeeZA**
Idea: consider using `molehill mountain` since it's a free fight

### CODE REVIEW

File: src/fights.ts [hunk 1]
```diff
@@ -2096,6 +2097,7 @@ export function deliverThesisIfAble(): void {
   safeRestore();
 
   if (
+    myBuffedstat($stat`Muscle`) < 1800 &&
```
Comment (eegeeZA) [inline src/fights.ts:2100]: Matches the muscle stat lower down. Should I extract to a local `const`?
https://github.com/Loathing-Associates-Scripting-Society/garbage-collector/blob/acf0323471e4fd54c6e028db77faa0718c839426/src/fights.ts#L2110

Comment (horrible-little-slime) [reply src/fights.ts:2100]: yeah, good shout

**Comment by horrible-little-slime** [comment]
does molehill mountain scale high enough?

we should maximize muscle before we check for triple-sized as well

File: src/fights.ts (hunk 1 above)
Comment (eegeeZA) [reply src/fights.ts:2100]: Cool, going to refactor a lot more. Based on the comment about maximising muscle before triple-sized.

**Comment by eegeeZA** [comment]
Based on my calc's moleman does scale and required lower Muscle. Check my calc and let me know:
`(849+15) * 1.5 = 1296 = min(2 * floor(1296^0.25), 11)`

**Comment by docrostov** [comment]
your calculation looks right, molehill mountain should theoretically work

**Comment by eegeeZA** [comment]
The code will now prioritise fighting Moleman if available. The required muscle has also been optimised to only require the minimum, depending on which monster will be fought. And lastly, the Powerful Glove will not be used if one already has enough buffed Muscle.

**Comment by horrible-little-slime** [comment]
Gotta delete that unused import

**Comment by eegeeZA** [comment]
> Gotta delete that unused import

Sorted. Apologies for all the quick successive pushes.

File: src/fights.ts [hunk 2]
```diff
@@ -716,7 +721,7 @@ const freeFightSources = [
     }
   ),
   new FreeFight(
-    () => (have($item`molehill mountain`) && !get("_molehillMountainUsed") ? 1 : 0),
+    () => (molemanReady() && !thesisReady() ? 1 : 0),
```
Comment (horrible-little-slime) [inline src/fights.ts:724]: Based on the way thesisReady works, people who don't own a prof will never fight the moleman.

Comment (eegeeZA) [reply src/fights.ts]: You're quite right. I've adjusted the logic.

Comment (eegeeZA) [reply src/fights.ts]: Hang on, isn't `(thesisReady() ? 0 : molemanReady() ? 1 : 0)` the same as `(molemanReady() && !thesisReady() ? 1 : 0)`? How will people without a Prof be blocked from the fight?
| Thesis | Moleman  | desired |
|--------|-------|---------|
| TRUE   | TRUE  | 0       |
| TRUE   | FALSE | 0       |
| FALSE  | TRUE  | 1       |
| FALSE  | FALSE | 0       |

**Review by horrible-little-slime — APPROVED** [review]

File: src/fights.ts (hunk 2 above)
Comment (horrible-little-slime) [reply src/fights.ts]: oh, I just straight-up misread that then. You're right!

Comment (eegeeZA) [reply src/fights.ts]: Hehe cool. The refactored version probably reads a bit easier. Thanks for the approve.

**Pull request merged**